bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java

```diff
@@ -170,7 +175,51 @@ private void handleCommandInternal(ChannelUID channelUID, Command command)
                         updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE, "Rebooting");
                         break;
                 }
+                break;
+            case RECORD_INPUT_CHANNEL:
+                recordDeviceInput(command);
+                break;
+            case RECORDED_INPUT_CHANNEL:
+                String recordName = getRecordPropertyName(command);
+                var inputCommand = this.getThing().getProperties().get(recordName);
+                if (inputCommand != null) {
+                    adbConnection.sendInputEvents(inputCommand);
+                }
+                break;
+        }
+    }
+
+    private void recordDeviceInput(Command recordNameCommand)
+            throws AndroidDebugBridgeDeviceException, InterruptedException, TimeoutException, ExecutionException {
+        var recordName = recordNameCommand.toFullString();
+        if (!RECORD_NAME_PATTERN.matcher(recordName).matches()) {
+            logger.warn("Invalid record name, accepts alphanumeric values with '_'.");
+            return;
         }
+        String recordPropertyName = getRecordPropertyName(recordName);
+        logger.debug("RECORD: {}", recordPropertyName);
+        if (!recordPropertyName.isEmpty() && !recordPropertyName.equals("NULL")) {
+            var eventCommand = adbConnection.recordInputEvents();
+            if (eventCommand.isEmpty()) {
+                logger.debug("No events recorded");
+                if (this.getThing().getProperties().containsKey(recordPropertyName)) {
+                    this.getThing().setProperty(recordPropertyName, null);
+                    updateProperties(editProperties());
```

Comment:
Why not using updateProperty rather than these people lines?

Reply:
Looking at core framework code, I have the feeling that it should work even with a null value. You can try ignoring the compiler warning and see if it works. 
Other and maybe better option would be to set the value to an empty string.

Reply:
It gives me a compiler error when I pass a null value to the updateProperty fn. And if I pass and empty string, the property doesn't get removed which is what I want.
So I can not find other way to remove the property.

Reply:
Even removing the property instead of setting if to null doesn't work. Setting it to null seems to be the only way to get it removed from the thing correctly.

Reply:
@lolodomo, Is it ok to leave it as is?

Reply:
No at least until I understand and validate your strange code and understand a simple and normal call is not possible. 
I will download your branch and see.

Reply:
Sure.
I think the problem itself is in the BaseThingHandler class it makes no sense to make the value non nullable on the updateProperty function and all the related ones when the updateProperty method in the thing accepts null. 
So I'm having to bypass this by calling the editProperties method to get and instance of a non nullable map which contains in fact a null value, or at least that's what I think is happening.
Either way I'll open and issue on the core with this info to see if I'm missing something.
Let me know if you find something and thanks for the review.

Reply:
Why not simply removing the property? 
It makes no real sense to set a property to NULL.

Reply:
I already tried that also but it makes non effect. 
But event if that had worked, I need to call the updateProperties/updateProperty function for the changes to be persisted because internal they notify the binding core that there is property changes. That's what I discover after I saw my the properties disappear after a reboot.

Reply:
I discovered that I have to set the properties to null to delete them reading the forum, documentation or a comment on other binding, I can't remember, I agree it's weird. A deleteProperty method on the BaseThingHandler class will be great.

Reply:
My idea was to use the code you proposed in your core issue but you're right, it cannot work. 
The solution would be to add a method `removeProperty` in the core framework. 
So I see no proper solution now. If your solution is working, that's fine for me.